src/components/shared/Input.js

```diff
@@ -1,89 +1,54 @@
-  const [passwordMaxError, SetPasswordMaxError] = useState(false)
-  const passwordUpperBound = 16
+export default function Input({
+  label,
+  theme,
+  inputState,
+  setInputState,
+  inputType = 'text',
+  upperBound = 16,
+  id = '',
+}) {
+  const [inputMinError, SetInputMinError] = useState(false)
+  const [inputMaxError, SetInputMaxError] = useState(false)
+  const [someInputTyped, SetSomeInputTyped] = useState(false)
 
   useEffect(() => {
-    if (demo1.password.length === passwordUpperBound) {
-      SetPasswordMaxError(true)
-    } else {
-      SetPasswordMinError(false)
-      SetPasswordMaxError(false)
+    if (inputState.length > 0) {
+      SetSomeInputTyped(true)
     }
-  }, [demo1.password])
-
-  const passwordChange = (e) => {
-    e.preventDefault()
-    update({
-      demo1: {
-        ...demo1,
-        password: e.target.value,
-      },
-    })
-  }
-
-  const enterPassword = (e) => {
-    e.preventDefault()
-    if (demo1.password.length === 0) {
-      SetPasswordMinError(true)
-    } else if (demo1.password.length > 0) {
-      update({
-        demo1: {
-          ...demo1,
-          isPasswordSet: true,
-        },
-      })
-    }
-  }
+    SetInputMaxError(inputState.length === upperBound)
+    SetInputMinError(inputState.length === 0 && someInputTyped)
+  }, [inputState, someInputTyped, upperBound])
 
   return (
-    <Container styles={{ height: '100%', paddingTop: '100px' }} size={10}>
-      <form>
-        <Label {...password.label}>insert your password</Label>
-        <Row>
-          <PasswordInput
-            {...password.input}
-            suggested={'shhhh-secret'}
-            id={'Password'}
-            type={'password'}
-            maxLength={passwordUpperBound}
-            onChange={(e) => {
-              passwordChange(e)
-            }}
-          />
-          <Button
-            {...password.button}
-            id={'submit-password'}
-            onClick={(e) => {
-              enterPassword(e)
-            }}
-          />
-        </Row>
-        {passwordMinError && (
-          <Warning {...password.warning}>
-            Password cannot be empty, must be a <br />
-            maximum of {passwordUpperBound} characters
-          </Warning>
-        )}
-        {passwordMaxError && (
-          <Warning {...password.warning}>
-            Maximum password length reached
-          </Warning>
-        )}
-      </form>
-    </Container>
+    <div>
+      <Label {...theme.label}>{label}</Label>
+      <Row>
+        <InputBox
+          {...theme.input}
+          id={id}
+          type={inputType}
+          maxLength={upperBound}
+          onChange={(e) => {
+            setInputState(e.target.value)
+          }}
+        />
+      </Row>
+      <Warning
+        visibility={inputMinError || inputMaxError ? 'visible' : 'hidden'}
+        {...theme.warning}
+      >
+        {inputMinError && <>Cannot be empty</>}
+        {inputMaxError && <>Maximum length reached</>}
```

Comment:
I wonder if we'd be better off elevating the evaluation of whether an input is eroneous to the calling component rather than here. Perhaps have an `inputError` prop and that way all input validation is externalised

Reply:
Hopefully the new commit I've just added does what you're saying?